### PR TITLE
[vecsim] memory report and new info 

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

**Review by ashtul — COMMENTED** [review]
Looks good!
Can we add a test that will check insertion and deletion stats updated?

**Comment by DvirDukhan** [comment]
@ashtul I didn't finish yet :)

File: src/document.c [hunk 1]
```diff
@@ -515,9 +515,9 @@ FIELD_BULK_INDEXER(vectorIndexer) {
     }
   }
   // TODO: change return value to NRN_AddRv
-  int rv = VecSimIndex_AddVector(rt, fdata->vector, aCtx->doc->docId);
   // TODO: update size statistics but put in a separate field to distinguise from inverted indexes
   // ctx->spec->stats.invertedSize += rt->size * sizeof(double) * 2;
```
Comment (ashtul) [inline src/document.c:519]: remove comments

File: src/vector_index.c [hunk 2]
```diff
@@ -47,7 +47,19 @@ static VecSimIndex *openVectorKeysDict(RedisSearchCtx *ctx, RedisModuleString *k
   // create new vector data structure
   kdv = rm_calloc(1, sizeof(*kdv));
   kdv->p = VecSimIndex_New(&fieldSpec->vecSimParams);
-  dictAdd(ctx->spec->keysDict, keyName, kdv);
+  VecSimIndexInfo indexInfo = VecSimIndex_Info(kdv->p);
+  switch (indexInfo.algo)
+  {
+    case VecSimAlgo_BF:
+      spec->stats.invertedSize = indexInfo.bfInfo.memory;
```
Comment (ashtul) [inline src/vector_index.c:54]:
```suggestion
      spec->stats.invertedSize += indexInfo.bfInfo.memory;
```

File: src/vector_index.c [hunk 3]
```diff
@@ -47,7 +47,19 @@ static VecSimIndex *openVectorKeysDict(RedisSearchCtx *ctx, RedisModuleString *k
...
+      break;
+    case VecSimAlgo_HNSWLIB:
+      spec->stats.invertedSize = indexInfo.hnswInfo.memory;
```
Comment (ashtul) [inline src/vector_index.c:57]:
```suggestion
      spec->stats.invertedSize += indexInfo.hnswInfo.memory;
```

Comment (ashtul) [reply src/vector_index.c:57]: `spec->stats` is a for all fields.

Comment (ashtul) [reply src/vector_index.c:57]: @MeirShpilraien @DvirDukhan  WDYT about adding `VectorSize` field to `IndexStats`?
This isn't an inverted index structure and might mislead users.

**Review by ashtul — COMMENTED** [review]
@DvirDukhan a few small comments

File: tests/pytests/test_vecsim.py [hunk 4]
```diff
@@ -327,3 +331,96 @@ def test_with_fields(env):
...
+def get_vecsim_memory(env, index_key, field_name):
+    return float(to_dict(env.cmd("FT.DEBUG", "VECSIM_INFO", index_key, field_name))["MEMORY"])/0x100000
+
+
+def test_memory_info(env):
+    # Skip on cluster as FT.DEBUG not supported.
+    env.skipOnCluster()
+    # This test flow adds two vectors and deletes them. The test checks for memory increase in Redis and RediSearch upon insertion and decrease upon delete.
+    conn = getConnectionByEnv(env)
+    conn = getConnectionByEnv(env)
```
Comment (ashtul) [inline tests/pytests/test_vecsim.py:345]: remove

**Review by ashtul — APPROVED** [review]
Looks good.

**Pull request merged**